sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -431,7 +434,7 @@ pi_result _pi_queue::signalEventFromCmdListIfLastEventDiscarded(
   pi_event Event;
   PI_CALL(createEventAndAssociateQueue(
       this, &Event, PI_COMMAND_TYPE_USER, CommandList,
-      /* IsDiscarded */ false, /* ForceHostVisible */ false))
+      /* IsInternal */ true, /* ForceHostVisible */ false))
```

Comment:
Unfortunately, I think this change will cause failures. It was intentionally made non-internal because we don't want to reset and reuse events which connect different command lists because it will lead to situation like this:
cmdlist1
	signal event1

cmdlist2
	barrier wait event1

cmdlist3
	barrier wait event1

cmdlist2 and cmdlist3 may start simultaneously (for example, if they are in compute and copy engine respectively) which is not correct. This was the reason why we always wanted to have new event to connect command lists - that's why such kind of events had IsInternal = false. I.e. currently IsInternal = true also means that event  can be reset and reused.

Reply:
> currently IsInternal = true also means that event can be reset and reused.

Then this is a problem. These should be separate properties:
1) Internal meaning that it has no uses from SYCL RT
2) Re-usable in in-order queue with discard events

Right? Any preference how to easily separate them?

Reply:
Yes, I agree that we need to separate those properties. I don't have any good ideas how to do this except having one more flag.

Reply:
Actually, we may consider replacing IsInternal with just checking "ZeEvent == nullptr".

Reply:
OK. I think I've done it without yet another boolean argument, please take a look.

Reply:
Thank you!